tests/test_functions.py

```
@@ -0,0 +1,137 @@
+    return None
+
+def is_valid_type(actual_type, expected_type):
+    if isinstance(expected_type, Type):
+        return actual_type & expected_type
+    return actual_type & expected_type.__args__[0]
+
+def validate_function(func, expected_args_type):
+    args_type = []
+    for _ in range(len(expected_args_type)):
+        args_type.append(all_types)
+    for args_type in product(*args_type):
```

Ruff (0.11.9)

Loop control variable `args_type` overrides iterable it iterates

(B020)